pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/MeasurementsTNCuda.hpp

```diff
@@ -51,13 +68,98 @@ template <class TensorNetT> class MeasurementsTNCuda {
 
     const TensorNetT &tensor_network_;
 
   public:
     explicit MeasurementsTNCuda(const TensorNetT &tensor_network)
         : tensor_network_(tensor_network){};
 
+    /**
+     * @brief Probabilities for a subset of the full system.
+     *
+     * @tparam thread_per_block Number of threads per block in the CUDA kernel
+     * and is default as `256`. `256` is chosen as a default value because it is
+     * a balance of warp size and occupancy. Note that this number is not
+     * optimal for all cases and may need to be adjusted based on the specific
+     * use case, especially the number of elements in the subset is small.
+     *
+     * @param wires Wires will restrict probabilities to a subset
+     * of the full system.
+     * @param numHyperSamples Number of hyper samples to be used in the
+     * calculation and is default as 1.
+     *
+     * @return Floating point std::vector with probabilities.
+     */
+    template <std::size_t thread_per_block = 256>
+    auto probs(const std::vector<std::size_t> &wires,
+               const int32_t numHyperSamples = 1) -> std::vector<PrecisionT> {
+        PL_ABORT_IF_NOT(std::is_sorted(wires.begin(), wires.end()),
+                        "Invalid wire indices order. Please ensure that the "
+                        "wire indices are in the ascending order.");
+
+        const std::size_t length = std::size_t{1} << wires.size();
+
+        std::vector<PrecisionT> h_res(length, 0.0);
+
+        DataBuffer<CFP_t, int> d_output_tensor(
+            length, tensor_network_.getDevTag(), true);
+
+        d_output_tensor.zeroInit();
+
+        DataBuffer<PrecisionT, int> d_output_probs(
+            length, tensor_network_.getDevTag(), true);
+
+        tensor_network_.get_state_tensor(d_output_tensor.getData(),
+                                         d_output_tensor.getLength(), wires,
+                                         numHyperSamples);
+
+        // `7` here means `256` elements to be calculated
+        // LCOV_EXCL_START
+        if (wires.size() > 7) {
+            getProbs_CUDA(d_output_tensor.getData(), d_output_probs.getData(),
+                          length, static_cast<int>(thread_per_block),
+                          tensor_network_.getDevTag().getStreamID());
+
+            PrecisionT sum;
+
+            asum_CUDA_device<PrecisionT>(
+                d_output_probs.getData(), length,
+                tensor_network_.getDevTag().getDeviceID(),
+                tensor_network_.getDevTag().getStreamID(),
+                tensor_network_.getCublasCaller(), &sum);
+
+            PL_ABORT_IF(sum == 0.0, "Sum of probabilities is zero.");
+
+            normalizeProbs_CUDA(d_output_probs.getData(), length, sum,
+                                static_cast<int>(thread_per_block),
+                                tensor_network_.getDevTag().getStreamID());
+
+            d_output_probs.CopyGpuDataToHost(h_res.data(), h_res.size());
+        } else {
```

Comment:
Fewer wires should result in small runtimes in any case, so do we really need this branch?

Reply:
Could we benchmark the host vs. device branch for 2 < num_wires < 8 ?

Reply:
Some updates: cpu computation (`around 0.02 ms`) will be faster than A100 (`around 0.07ms`) if the number of wires is less than `10`. As @vincentmr said, it's minor.

Reply:
Note that we're comparing the naive CPU implementation with GPU ;) But if the entire computation is in order of 0.0x milliseconds, this minor difference makes sense.